### PR TITLE
Use clang++ front end for linking Evergreen builds.

### PR DESCRIPTION
**Pull request by y4vor**
- In the past there was an issue passing arguments from the clang++ to lld. Looks like there issue is gone now and we can use clang++ in the same way as we do on all the other platform.

b/206642994

### CODE REVIEW

File: starboard/build/toolchain/linux/BUILD.gn [hunk 1]
```diff
@@ -37,6 +36,3 @@ clang_toolchain("x86") {
     is_clang_16 = true
   }
 }
-
-clang_lld_toolchain("clang_lld") {
```
Comment (andrewsavage1) [inline starboard/build/toolchain/linux/BUILD.gn:41]: This is used for at least the modular build so we'll probably have to create another basic clang toolchain without explicit os/cpu references that BUILDCONFIG can point to for modular build

**Review by andrewsavage1 — COMMENTED** [review]
Thanks for this awesome cleanup!

File: starboard/build/toolchain/linux/BUILD.gn (hunk 1 above)
Comment (y4vor) [reply starboard/build/toolchain/linux/BUILD.gn:41]: But isn't it used because of the way Evergreen setup the flags?
Do we really need to use it?

Comment (andrewsavage1) [reply starboard/build/toolchain/linux/BUILD.gn]: We don't want an lld toolchain, but we want a general clang toolchain that doesn't set any os/cpu/config path etc., just the clang_base_path

Comment (y4vor) [reply starboard/build/toolchain/linux/BUILD.gn]: Yeah agree. The flags are shared.

File: starboard/build/toolchain/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build/config/clang/clang.gni")
+import("//build/toolchain/gcc_toolchain.gni")
+
+clang_toolchain("clang") {
```
Comment (andrewsavage1) [inline starboard/build/toolchain/BUILD.gn:18]: Let's explicitly set `clang_base_path` in here

Comment (y4vor) [reply starboard/build/toolchain/BUILD.gn:18]: Done

**Review by hlwarriner — APPROVED** [review]
Nice!

**Review by andrewsavage1 — APPROVED** [review]

**Pull request merged**